### PR TITLE
fix(build): #1002 args manage ports fix

### PR DESCRIPTION
**Pull request by este6an13**
- default import of `shellCommands` builtin in `makeTemplate`

### CODE REVIEW

File: src/args/manage-ports/template.sh [hunk 1]
```diff
@@ -50,7 +50,7 @@ function wait_for_tcp {
       error "Timeout while waiting for ${host}:${port} to open" \
         && return 1
     else
-      info "Waiting 1 second for ${host}:${port} to open, ${elapsed} seconds in total" \
+      info "Waiting 1 second for ${host}:${port} to open: ${elapsed} seconds in total" \
```
Comment (dsalaza4) [inline src/args/manage-ports/template.sh:53]: Such attention to detail! ❤️

**Comment by dsalaza4** [comment]
`m . /lintGitCommitMsg` failing. Please read https://makes.fluidattacks.com/contributing/#code-contributions

**Comment by dsalaza4** [comment]
@este6an13,

I just realized this is not a valid change 😢, here is why:

`makeDerivation` and `makeEntrypoint` both receive a `sh` script and execute it, this is why sourcing `__shellCommands__ ` by default on both of them makes sense.

`makeTemplate` is different. It receives any text file and simply replaces placeholders declared within it with `__argName__` values provided within the builtin. Here is an example of a `makeTemplate` using a simple txt file:

https://github.com/fluidattacks/makes/blob/91ee9fef013a9e3ad31fa4aac504eb62fd9639d1/makes/tests/makeTemplate/main.nix#L11-L19

This basically means that we cannot source `__shellCommands__` by default for `makeTemplate`.

After running some tests on this change I realized that the resulting templated script still cannot find the Makes' `info`, `warn` and `error` functions. Here's what I did:

```nix
{makeTemplate, ...}:
makeTemplate {
  replace = {
    __argTest__ = "This is a test";
  };
  name = "test-make-template-2";
  template = "error __argTest__";
}
```

The resulting template file was simply

```txt
error This is a test
```

When executing this with a shell interpreter like bash, It could not find `error` as it did not source it in the first place.

```sh
$ bash /nix/store/b2dk4p5gyarrgwd5gqd91vygi1dbnbxg-make-template-for-test-make-template-2/template
/nix/store/b2dk4p5gyarrgwd5gqd91vygi1dbnbxg-make-template-for-test-make-template-2/template: line 1: error: command not found
```

The only feasible solution I see here if we want to keep the `info`, `warn` and `error` functions is directly sourcing them in `managePorts` by:

adding `__shellCommands__` as a `replace` argument here:

https://github.com/fluidattacks/makes/blob/91ee9fef013a9e3ad31fa4aac504eb62fd9639d1/src/args/manage-ports/default.nix#L6-L15

and sourcing it here, right before declaring the functions:

https://github.com/fluidattacks/makes/blob/91ee9fef013a9e3ad31fa4aac504eb62fd9639d1/src/args/manage-ports/template.sh#L1-L3

**Comment by dsalaza4** [comment]
BTW, let's keep this PR open for easier traceability, you can simply amend your commit and push again, that will update the PR.

File: src/args/manage-ports/template.sh [hunk 2]
```diff
@@ -50,11 +52,12 @@ function wait_for_tcp {
...
         && sleep 1 \
         && elapsed="$(("${elapsed}" + 1))" \
         && continue
     fi
+    break
```
Comment (dsalaza4) [inline src/args/manage-ports/template.sh:60]: I think this line breaks the cycle in the first iteration?

**Comment by este6an13** [comment]
Hello @dsalaza4, I made a few changes following your recommendations. 

I'm passing `__shellCommands__` as a `replace` argument in `managePorts`. Then I'm sourcing it in `manage-ports/template.sh`. 

I tried to test this by modifying the files in `makes/makes/tests/makeTemplate` folder, and it worked for the default `template.txt` file just by using this new `__argShellCommands__` argument there. However when I tried to do this for a new file, I got the following error:

image

I'm just showing the end of the backtrace. It says that it cannot find my `my-template.sh` file. I executed `m . /tests/makeTemplate` from the root `makes` folder.

Here's how I modified the `main.nix` file:

```nix
{
  fromYaml,
  makeTemplate,
  __shellCommands__,
  ...
}: let
  testFile = fromYaml (
    builtins.readFile ./test.yaml
  );
  testString = testFile.testTitle;
in
  makeTemplate {
    replace = {
      __argShellCommands__ = __shellCommands__;
    };
    name = "my-template";
    template = ./my-template.sh;
  }
```
And this is the new `my-template.sh` file I created:

``` bash
source __argShellCommands__

error "Hello"
````
Could you help me understanding why this happens? 

I'm looking forward to your comments. Thank you!

File: src/args/manage-ports/template.sh (hunk 2 above)
Comment (este6an13) [reply src/args/manage-ports/template.sh:60]: That's right, my mistake. I added that line for debugging. Will remove it now.

**Comment by dsalaza4** [comment]
> Hello @dsalaza4, I made a few changes following your recommendations.
> 
> I'm passing `__shellCommands__` as a `replace` argument in `managePorts`. Then I'm sourcing it in `manage-ports/template.sh`.
> 
> I tried to test this by modifying the files in `makes/makes/tests/makeTemplate` folder, and it worked for the default `template.txt` file just by using this new `__argShellCommands__` argument there. However when I tried to do this for a new file, I got the following error:
> 
> image
> I'm just showing the end of the backtrace. It says that it cannot find my `my-template.sh` file. I executed `m . /tests/makeTemplate` from the root `makes` folder.
> 
> Here's how I modified the `main.nix` file:
> 
> ```nix
> {
>   fromYaml,
>   makeTemplate,
>   __shellCommands__,
>   ...
> }: let
>   testFile = fromYaml (
>     builtins.readFile ./test.yaml
>   );
>   testString = testFile.testTitle;
> in
>   makeTemplate {
>     replace = {
>       __argShellCommands__ = __shellCommands__;
>     };
>     name = "my-template";
>     template = ./my-template.sh;
>   }
> ```
> 
> And this is the new `my-template.sh` file I created:
> 
> ```shell
> source __argShellCommands__
> 
> error "Hello"
> ```
> 
> Could you help me understanding why this happens?
> 
> I'm looking forward to your comments. Thank you!

I think this is because you did not `git add` the new files? Read this https://makes.fluidattacks.com/contributing/#testing-your-local-changes

**Comment by dsalaza4** [comment]
> > Hello @dsalaza4, I made a few changes following your recommendations.
> > I'm passing `__shellCommands__` as a `replace` argument in `managePorts`. Then I'm sourcing it in `manage-ports/template.sh`.
> > I tried to test this by modifying the files in `makes/makes/tests/makeTemplate` folder, and it worked for the default `template.txt` file just by using this new `__argShellCommands__` argument there. However when I tried to do this for a new file, I got the following error:
> > image
> > I'm just showing the end of the backtrace. It says that it cannot find my `my-template.sh` file. I executed `m . /tests/makeTemplate` from the root `makes` folder.
> > Here's how I modified the `main.nix` file:
> > ```nix
> > {
> >   fromYaml,
> >   makeTemplate,
> >   __shellCommands__,
> >   ...
> > }: let
> >   testFile = fromYaml (
> >     builtins.readFile ./test.yaml
> >   );
> >   testString = testFile.testTitle;
> > in
> >   makeTemplate {
> >     replace = {
> >       __argShellCommands__ = __shellCommands__;
> >     };
> >     name = "my-template";
> >     template = ./my-template.sh;
> >   }
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > And this is the new `my-template.sh` file I created:
> > ```shell
> > source __argShellCommands__
> > 
> > error "Hello"
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Could you help me understanding why this happens?
> > I'm looking forward to your comments. Thank you!
> 
> I think this is because you did not `git add` the new files? Read this https://makes.fluidattacks.com/contributing/#testing-your-local-changes

I just ran the same test and it worked well for me!

**Comment by dsalaza4** [comment]
Please add missing mailmap entry por noreply https://github.com/fluidattacks/makes/actions/runs/5537342039/jobs/10106243030?pr=1120

**Review by dsalaza4 — APPROVED** [review]
LGTM!

**Pull request merged**